velox/functions/lib/Re2Functions.cpp

```diff
@@ -300,6 +342,43 @@ class Re2SearchAndExtract final : public VectorFunction {
   }
 };
 
+class LikeConstantPattern final : public VectorFunction {
+ public:
+  explicit LikeConstantPattern(StringView pattern, char escapeChar)
+      : re_(toStringPiece(likePatternToRe2(pattern, escapeChar)), RE2::Quiet) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      Expr* /* caller */,
+      EvalCtx* context,
+      VectorPtr* resultRef) const final {
+    VELOX_CHECK(args.size() == 2 || args.size() == 3);
+
+    // apply() will not be invoked if the selection is empty.
+    checkForBadPattern(re_);
+
+    exec::LocalDecodedVector toSearch(context, *args[0], rows);
+    // TODO: Potentially re-use the string vector, not just the buffer.
+    FlatVector<StringView>& result =
```

Comment:
Shouldn't the result type be a boolean?

Reply:
I misread the PrestoDB docs to interpret "like" as a vector function that returned the original column values. But a boolean operator does seem more preferable. Have changed this.